src/System.Text.Json/src/System/Text/Json/Serialization/Converters/DefaultConverters.cs

```diff
@@ -2,12 +2,27 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System.Collections.Concurrent;
 using System.Reflection;
 
 namespace System.Text.Json.Serialization.Converters
 {
     internal static class DefaultConverters
     {
+        private static readonly ConcurrentDictionary<Type, object> _valueConverters = new ConcurrentDictionary<Type, object>();
+
+        static DefaultConverters()
+        {
+            _valueConverters.TryAdd(typeof(DateTimeOffset), new JsonValueConverterDateTimeOffset());
+            _valueConverters.TryAdd(typeof(Guid), new JsonValueConverterGuid());
+            _valueConverters.TryAdd(typeof(JsonElement), new JsonValueConverterJsonElement());
+        }
```

Comment:
Does anything change this dictionary after it's initialized?  It looks like the answer is "no", in which case it'd be better to just make it a Dictionary rather than a ConcurrentDictionary.

It'd also be better to not use an explicit static cctor, which can add overhead on every static access.

And, nit, the static field should have an s_ prefix.

So, it'd be better for this to be something like:
```C#
private static readonly Dictionary<Type, object> s_valueConverters = CreateValueConverters();

private static Dictionary<Type, object> CreateValueConverters()
{
    var d = new Dictionary<Type, object>(3);
    d.Add(typeof(DateTimeOffset), new JsonValueConverterDateTimeOffset());
    d.Add(typeof(Guid), new JsonValueConverterGuid());
    d.Add(typeof(JsonElement), new JsonValueConverterJsonElement());
    return d;
}
```

Reply:
The initial thought was that we need a place to register user converters and this was just a step towards that. Thinking about it a little bit more that probably isn't the best choice- I doubt we'd want to have custom ones globally registered.

In any case I'll move it off the static constructor. Thanks for clarifying the [`beforefieldinit`](https://csharpindepth.com/Articles/BeforeFieldInit) issue. :)